receiver/dockerstatsreceiver/metrics.go

```diff
@@ -0,0 +1,397 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerstatsreceiver
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	dtypes "github.com/docker/docker/api/types"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+)
+
+const (
+	metricPrefix = "container."
+)
+
+// client.ContainerInspect() response container
+// stats and translated environment string map
+// for potential labels
+type DockerContainer struct {
+	*dtypes.ContainerJSON
+	EnvMap map[string]string
+}
+
+func ContainerStatsToMetrics(
+	containerStats *dtypes.StatsJSON,
+	container *DockerContainer,
+	config *Config,
+) (*consumerdata.MetricsData, error) {
+	now, _ := ptypes.TimestampProto(time.Now())
+
+	var metrics []*metricspb.Metric
+	metrics = append(metrics, blockioMetrics(&containerStats.BlkioStats, now, config.ProvideAllBlockIOMetrics)...)
+	metrics = append(metrics, cpuMetrics(&containerStats.CPUStats, &containerStats.PreCPUStats, now, config.ProvideAllCPUMetrics)...)
+	metrics = append(metrics, memoryMetrics(&containerStats.MemoryStats, now, config.ProvideAllMemoryMetrics)...)
+	metrics = append(metrics, networkMetrics(&containerStats.Networks, now, config.ProvideAllNetworkMetrics)...)
+
+	if len(metrics) == 0 {
+		return nil, nil
+	}
+
+	md := &consumerdata.MetricsData{
+		Metrics: metrics,
+		Resource: &resourcepb.Resource{
+			Type: "docker",
+			Labels: map[string]string{
+				"container.hostname": container.Config.Hostname,
+				"container.id":       container.ID,
+				"container.image":    container.Config.Image,
+				"container.name":     strings.TrimPrefix(container.Name, "/"),
```

Comment:
Let's use constants from core for these labels since they're available [here](https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go#L27-L29).

I believe the Type should be set to `container` as per https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/container.md.

Reply:
Updated